beacon-chain/casper/sharding_test.go

```diff
@@ -116,11 +120,136 @@ func TestSampleAttestersAndProposers(t *testing.T) {
 		t.Errorf("Get attesters failed, expected: %v got: %v", validatorList[:len(attesters)], attesters)
 	}
 
-	indices, err = ValidatorsByHeightShard(common.Hash{'A'}, validators, 1, 0)
+	indices, err := ShuffleValidatorsToCommittees(common.Hash{'A'}, validators, 1, 0)
 	if err != nil {
 		t.Errorf("validatorsByHeightShard failed with %v:", err)
 	}
-	if len(indices) != 8192 {
-		t.Errorf("incorret length for validator indices. Want: 8192. Got: %v", len(indices))
+	if len(indices) != params.CycleLength {
+		t.Errorf("incorret length for validator indices. Want: %d. Got: %d", params.CycleLength, len(indices))
+	}
+}
+
+func TestGetCommitteeParams1(t *testing.T) {
+	numValidators := params.CycleLength * params.MinCommiteeSize / 4
+
+	committesPerSlot, slotsPerCommittee := getCommitteeParams(numValidators)
+	if committesPerSlot != 1 {
+		t.Fatalf("Expected committeesPerSlot to equal %d: got %d", 1, committesPerSlot)
+	}
+
+	if slotsPerCommittee != 4 {
+		t.Fatalf("Expected slotsPerCommittee to equal %d: got %d", 4, slotsPerCommittee)
+	}
+}
+
+func TestGetCommitteeParams2(t *testing.T) {
+	numValidators := params.CycleLength * params.MinCommiteeSize
+
+	committesPerSlot, slotsPerCommittee := getCommitteeParams(numValidators)
+	if committesPerSlot != 1 {
+		t.Fatalf("Expected committeesPerSlot to equal %d: got %d", 1, committesPerSlot)
+	}
+
+	if slotsPerCommittee != 1 {
+		t.Fatalf("Expected slotsPerCommittee to equal %d: got %d", 1, slotsPerCommittee)
+	}
+}
+
+func TestGetCommitteeParams3(t *testing.T) {
+	numValidators := params.CycleLength * params.MinCommiteeSize * 8
+
+	committesPerSlot, slotsPerCommittee := getCommitteeParams(numValidators)
+	if committesPerSlot != 5 {
+		t.Fatalf("Expected committeesPerSlot to equal %d: got %d", 5, committesPerSlot)
+	}
+
+	if slotsPerCommittee != 1 {
+		t.Fatalf("Expected slotsPerCommittee to equal %d: got %d", 1, slotsPerCommittee)
+	}
+}
+
+func TestValidatorsByHeightShard1(t *testing.T) {
```

Comment:
Rename test function names to be more descriptive. Such as TestValidatorsByHeightIncorrectLength or better names for debugging.